extensions/ql-vscode/src/query-status.ts

```diff
@@ -1,5 +1,23 @@
+import { assertNever } from './pure/helpers-pure';
+import { VariantAnalysisStatus } from './remote-queries/shared/variant-analysis';
+
 export enum QueryStatus {
   InProgress = 'InProgress',
   Completed = 'Completed',
   Failed = 'Failed',
 }
+
+export function variantAnalysisStatusToQueryStatus(status: VariantAnalysisStatus): QueryStatus {
+  switch (status) {
+    case VariantAnalysisStatus.Succeeded:
+      return QueryStatus.Completed;
+    case VariantAnalysisStatus.Failed:
+      return QueryStatus.Failed;
+    case VariantAnalysisStatus.InProgress:
+      return QueryStatus.InProgress;
+    case VariantAnalysisStatus.Canceled:
+      return QueryStatus.Completed;
```

Comment:
We did discuss this during the mobbing session and ended up deciding that `Canceled` would be `Completed`.

I'm fine with either option tbh. 

@robertbrignull, @norascheuch, @shati-patel any strong opinions?

Reply:
I think our argument for `Canceled` was that it's a conscious user action, as in, they have successfully halted the variant analysis when they wanted to 😅 Assuming at least one of the repo tasks succeeded, there will be some completed results to display.

`Failed` sounded more like "something went wrong".

(I can't see any specific designs for the canceled case though, so I'm not sure how we need to indicate this in the view/history 🤔 Perhaps as "completed" but with a warning message, rather than an error.)

Reply:
I agree with @shati-patel, that was our reasoning from the meeting.

I suppose we potentially also have the option to add a `Cancelled` option to `QueryStatus`. Showing as "completed but with a warning" sounds like a good compromise to me.

We could just go with either option for now and raise it as a point for design to look at and decide what to do in this case?